### PR TITLE
Update README and add help target to make

### PR DESCRIPTION
**Pull request by jphuynh**
Small update to include shellcheck in the README

Signed-off-by: Jean-Pierre Huynh <jean-pierre.huynh@ounet.fr>

### CODE REVIEW

**Comment by dnephin** [comment]
We don't include all the available tasks in the `README` these are just here as an example. 

Maybe shellcheck could be added to lint (`make -f docker.Makefile lint shellcheck`).

We could also add a `CONTRIBUTING` to document it further

**Comment by thaJeztah** [comment]
Perhaps we can add a `make help` to make the Makefile self-documenting, like we have in https://github.com/moby/moby/blob/master/Makefile#L134

**Comment by jphuynh** [comment]
I like the idea @thaJeztah 👍

**Comment by jphuynh** [comment]
I gave it a go. 

@dnephin I'm not sure about the wording for the `watch` target. Let me know if it sounds good.

File: README.md [hunk 1]
```diff
@@ -23,10 +23,10 @@ Build binaries for all supported platforms:
 $ make -f docker.Makefile cross
 ```
 
-Run all linting:
+Run all linting and shellcheck validation:
 
 ```
-$ make -f docker.Makefile lint
+$ make -f docker.Makefile lint shellcheck
```
Comment (thaJeztah) [inline README.md]: We can mention the `help` make target in the readme (to see a list of all make options, type make ......help)

Comment (jphuynh) [inline README.md]: Why didn't think about that? 😅  It makes more sense. Thanks.

File: README.md [hunk 2]
```diff
@@ -23,12 +23,18 @@ Build binaries for all supported platforms:
 $ make -f docker.Makefile cross
 ```
 
-Run all linting:
+Run all linting and shellcheck validation:
```
Comment (dnephin) [inline README.md]: this comment doesn't match the code example anymore

File: Makefile [hunk 3]
```diff
@@ -3,62 +3,61 @@
...
 .PHONY: clean
-clean:
+clean: ## remove build artifacts
 	rm -rf ./build/* cli/winresources/rsrc_* ./man/man[1-9] docs/yaml/gen
 
-# run go test
-# the "-tags daemon" part is temporary
 .PHONY: test
-test:
+test: ## run go test (the "-tags daemon" part is temporary)
```
Comment (dnephin) [inline Makefile]: The note about `-tags daemon` is more for the reading of the file, not someone running tasks, can we remove that bit?

Comment (jphuynh) [inline Makefile]: Ok

File: README.md (hunk 2 above)
Comment (jphuynh) [inline README.md]: Oops :)

**Pull request merged**